pkg/trace/api/api.go

```diff
@@ -390,6 +400,39 @@ func (r *HTTPReceiver) rateLimited(n int64) bool {
 	return !r.RateLimiter.Permits(n)
 }
 
+// StatsProcessor implementations are able to process incoming client stats.
+type StatsProcessor interface {
+	// ProcessStats takes a stats payload and consumes it. It is considered to be originating
+	// from the given lang.
+	ProcessStats(p pb.ClientStatsPayload, lang string)
+}
+
+// handleStats handles incoming stats payloads.
+func (r *HTTPReceiver) handleStats(w http.ResponseWriter, req *http.Request) {
+	defer timing.Since("datadog.trace_agent.receiver.stats_process_ms", time.Now())
+
+	rd := NewLimitedReader(req.Body, r.conf.MaxRequestBytes)
+	slurp, err := ioutil.ReadAll(rd)
```

Comment:
Why read the data in a byte array to wrap it later in a bytes.NewReader?

Reply:
I guess it's a leftover from the previous change where we had multiple supported codecs. I'll change it.

Reply:
Done. I removed a test which makes sure the behaviour is correct if the payload is larger than 10MB. Realistically speaking that should technically be impossible since most payloads have 1-2 buckets and that helper code is already well tested. Should be good enough as far as I'm concerned.